vulpes/memory/gamestate/table.cpp

```diff
@@ -23,3 +23,21 @@ MemRef Table::new_at(MemRef id) {
 bool Table::remove(MemRef id) {
     return (datum_delete(this, id.raw) != 0xFFFFFFFF);
 }
+
+size_t Table::count() {
+    // Universal table element counting function.
+    // Made to work with any table.
+    size_t count = 0;
+    auto table = reinterpret_cast<GenericTable*>(this);
+    // Table array starts here.
+    uint32_t base = reinterpret_cast<uint32_t>(table->first);
```

Comment:
These names might be more descriptive. Up to you if you like them more.
base -> table_start_address
offset -> entry_offset

Also, vertical space is (mostly) free. Some white space would make this a little easier to parse